### PR TITLE
language: add full locale options support

### PR DESCRIPTION
**Pull request by cptMikky**
### Description

fixes https://github.com/rycee/home-manager/issues/1277

### Checklist

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/rycee/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

### CODE REVIEW

**Comment by cptMikky** [comment]
the `CONTRIBUTING` link in the issue template yields 404

File: modules/home-environment.nix [hunk 1]
```diff
@@ -40,11 +72,44 @@ let
...
+      name = mkOption {
         default = null;
         type = types.nullOr types.str;
         description = ''
-          The language to use for formatting times.
+          The language to use for personal names.
+        '';
+      };
+
+
```
Comment (rycee) [inline modules/home-environment.nix:83]:
```suggestion
```

File: modules/home-environment.nix [hunk 2]
```diff
@@ -40,11 +72,44 @@ let
...
+      measurement = mkOption {
+        default = null;
+        type = types.nullOr types.str;
+        description = ''
+          The language to use for measurement values.
+        '';
+      };
+
+      identification = mkOption {
```
Comment (rycee) [inline modules/home-environment.nix:108]: I think you can remove this one. As far as I can tell it's not something users are expected to set:

> LC_IDENTIFICATION - this is not a user-visible category, it contains information about the locale itself and is rarely useful for users or developers (but is listed here for completeness sake).

From https://sourceware.org/glibc/wiki/Locales#Overview

Comment (cptMikky) [reply modules/home-environment.nix]: you're right, I just took all the variables reported by `locale`

File: modules/home-environment.nix [hunk 3]
```diff
@@ -354,13 +411,29 @@ in
...
-        (maybeSet "LC_TIME" cfg.language.time);
+        (maybeSet "LC_NAME" cfg.language.name)
+        //
+        (maybeSet "LC_ADDRESS" cfg.language.address)
+        //
+        (maybeSet "LC_TELEPHONE" cfg.language.telephone)
+        //
+        (maybeSet "LC_MEASUREMENT" cfg.language.measurement)
+        //
+        (maybeSet "LC_IDENTIFICATION" cfg.language.identification);
```
Comment (rycee) [inline modules/home-environment.nix:436]: Need to remove this as well.

Comment (cptMikky) [reply modules/home-environment.nix]: :facepalm: of course, sorry

done.

**Comment by rycee** [comment]
Thanks! Rebased to master in 479274775f25be4000e41ee6584ea0865e77808b.

**Pull request closed without merging**